cmd/argocd-server/commands/argocd_server.go

```diff
@@ -160,6 +160,7 @@ func NewCommand() *cobra.Command {
 			stats.StartStatsTicker(10 * time.Minute)
 			stats.RegisterHeapDumper("memprofile")
 			argocd := server.NewServer(context.Background(), argoCDOpts)
+			argocd.Init(context.Background())
```

Comment:
This is not really your problem, but just noticed we call `context.Background()` three times in same function and you might consider consolidating to one ctx.

Reply:
This problem is fairly pervasive throughout `cmd/`.

I may look at this a little bit since it's an easy opportunity to get to know the codebase more.